feat(config): implement custom header field inside HostRules (#26225)
renovatebot/renovate: docs/usage/configuration-options.md

Enable got [http2](https://github.com/sindresorhus/got/blob/v11.5.2/readme.md#http2) support.

### headers

> You can provide `headers` object that includes fields to be forwarded to the HTTP request headers. By default, all headers starting with "X-" are allowed, see [allowedHeaders](self-hosted-configuration.md#allowedHeaders) to configure more permitted headers.

Reviewer: ```suggestion
You can provide `headers` object that includes fields to be forwarded to the HTTP request headers. By default, all headers starting with "X-" are allowed, a bot administrator may configure an override for [allowedHeaders](self-hosted-configuration.md#allowedHeaders) to configure more permitted headers.

Any `headers` value configured in bot admin `hostRules` (e.g. `config.js`) won't be validated so can contain any desired header regardless of `allowedHeaders`.
```

Example:

```json
{
  "hostRules": [
    {
      "matchHost": "https://domain.com/all-versions",
      "headers": {

>         "custom-header": "secret"

Reviewer: ```suggestion
        "X-custom-header": "secret"
```

Enable got [http2](https://github.com/sindresorhus/got/blob/v11.5.2/readme.md#http2) support.

### header

> You can provide `header` object that includes fields to be forwarded to the HTTP request header. By default, all header starting with "X-" are allowed, a bot administrator may configure an override for [allowedHeader](self-hosted-configuration.md#allowedHeader) to configure more permitted header.

Reviewer 2: ```suggestion
You can provide `header` object that includes fields to be forwarded to the HTTP request header. By default, all header starting with "X-" are allowed, a bot administrator may configure an override for [allowedHeader](./self-hosted-configuration.md#allowedHeader) to configure more permitted header.
```

> You can provide `header` object that includes fields to be forwarded to the HTTP request header. By default, all header starting with "X-" are allowed, a bot administrator may configure an override for [allowedHeader](./self-hosted-configuration.md#allowedHeader) to configure more permitted header.
> 
> Any `header` value configured in bot admin `hostRules` (e.g. `config.js`) won't be validated so can contain any desired header regardless of `allowedHeader`.

Reviewer 3: ```suggestion
You can provide a `header` object that includes fields to be forwarded to the HTTP request header.
By default, all headers starting with "X-" are allowed.

A bot administrator may configure an override for [`allowedHeader`](./self-hosted-configuration.md#allowedHeader) to configure more permitted headers.

`header` value(s) configured in the bot admin `hostRules` (for example in a `config.js` file) are _not_ validated, so it may contain any header regardless of `allowedHeader`.
```

Some small improvements to start with.

The docs for the `header` config option should explain _why_ somebody would want to override the default headers. It's not clear after reading this text what I as end-user gain from this option, when I should use it, when I should _not_ use it, if there's anything else I should know about.

> Example:

Reviewer 3: ```suggestion
For example:
```
